lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -601,8 +594,14 @@ class HtmlViewEmbedder {
       return;
     }
     final List<List<int>> overlayGroups = getOverlayGroups(_compositionOrder);
-    final List<int> viewsNeedingOverlays =
-        overlayGroups.map((List<int> group) => group.last).toList();
+    final List<int> viewsNeedingOverlays = overlayGroups
+        .where(
+          (List<int> group) => group.any(
+            (int viewId) => platformViewManager.isVisible(viewId)
+          )
+        )
+        .map((List<int> group) => group.last)
+        .toList();
```

Comment:
I don't know if this is on a performance hot path, but keep in mind that each `where`, `any`, and `map` cost an extra closure and a lazy-evaluated `Iterable` wrapper.

I'm also a little confused by the "overlay group" concept. If it's an overlay group, then presumably it already needs an overlay (or _is_ an overlay). Maybe these should be called `viewGoups` (and similarly the function `getOverlayGroups` be renamed to `getViewGroups`)?

Even better might be to introduce a `ViewGroup` class that has a `containsVisiblePlatformViews` boolean flag that's populated by `getOverlayGroups` as it's iterating over all views.

Reply:
I like the ViewGroup idea, let me see how much breakage I introduce :)

Reply:
I added a small OverlayGroup class that should help this bit of the code.